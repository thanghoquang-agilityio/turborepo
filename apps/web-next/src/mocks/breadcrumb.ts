export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
