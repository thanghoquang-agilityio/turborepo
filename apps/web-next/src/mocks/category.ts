export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
