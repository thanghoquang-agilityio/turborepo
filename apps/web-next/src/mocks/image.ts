import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants'

export const MOCK_IMAGE = {
  data: {
    id: '24',
    attributes: {
      name: 'default',
      url: SRC_IMAGE_NOT_AVAILABLE,
    },
  },
}

export const MOCK_LIST_IMAGE_RESPONSE = [
  {
    name: 'default1',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default2',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default3',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default4',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
]
