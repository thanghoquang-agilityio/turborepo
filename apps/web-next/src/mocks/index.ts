export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
