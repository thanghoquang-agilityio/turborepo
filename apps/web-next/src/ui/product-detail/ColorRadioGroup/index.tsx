import { Dispatch, SetStateAction, memo } from 'react'

// Components
import { Radio, RadioGroup, Tooltip } from '@nextui-org/react'

import { InView } from '@/components'
import { cn } from '@/lib/utils'
// Constants
import { COLOR_ITEMS, COLOR_ITEM_DEFAULT } from '@/constants'
import { StickIcon } from '@/icons'

interface ColorItems {
  color: string
}

export interface ColorRadioGroupProps {
  items: ColorItems[]
  valueColor: string
  setValueColor: Dispatch<SetStateAction<string>>
  additionalClass?: string
}

const ColorRadioGroup = memo(
  ({
    items,
    valueColor,
    setValueColor,
    additionalClass,
  }: ColorRadioGroupProps) => (
    <RadioGroup
      label='Choose a Color'
      orientation='horizontal'
      value={valueColor}
      onValueChange={setValueColor}
      classNames={{
        base: 'sm:gap-4 gap-2',
        wrapper: 'sm:gap-4 gap-2',
        label: 'text-base text-foreground-300 font-medium',
      }}
      className={additionalClass || ''}
    >
      {items.map(({ color }, index) => {
        const colorList = color.split('-')
        const isMultipleColor = colorList.length > 1

        const colorItem =
          COLOR_ITEMS.find((item) => item.color === color) || COLOR_ITEM_DEFAULT

        const isSelected = valueColor === color
        const borderColor = isMultipleColor
          ? 'border-danger-300'
          : `${colorItem.borderColor}`
        const backgroundColor = isMultipleColor
          ? 'bg-gradient-conic'
          : `${colorItem.backgroundColor} group-data-[hover-unselected=true]:${colorItem.backgroundColor}`
        const styleSelected =
          valueColor === color ? `${borderColor} p-0` : 'border-background-100'

        return (
          <Tooltip
            key={`tooltip-radio-${index}`}
            showArrow
            placement='top'
            content={color}
            classNames={{
              base: ['before:bg-primary-300'],
              content: [
                'py-2 px-4 shadow-xl',
                'text-background-100 bg-primary-300',
              ],
            }}
          >
            <Radio
              key={`color-item-${index}`}
              aria-label='Radio Color'
              value={color}
              className={`m-0 h-[54px] w-[54px] max-w-[64px] items-center justify-center rounded-full border-[3px] capitalize sm:h-[64px] sm:w-[64px] ${styleSelected}`}
              classNames={{
                wrapper: cn(
                  `inline-flex ${backgroundColor} sm:w-[50px] sm:h-[50px] w-[40px] h-[40px] hover:opacity-80 p-0 m-0`,
                  'cursor-pointer rounded-full border-none'
                ),
                control: 'bg-background-100 hidden',
                labelWrapper:
                  'absolute sm:top-[19px] top-[16px] sm:left-[10px] left-[6px] sm:w-6 sm:h-5 w-5 h-4',
              }}
            >
              {isSelected && (
                <InView>
                  <StickIcon customClass='text-background-100' />
                </InView>
              )}
            </Radio>
          </Tooltip>
        )
      })}
    </RadioGroup>
  )
)

ColorRadioGroup.displayName = 'ColorRadioGroup'
export default ColorRadioGroup
