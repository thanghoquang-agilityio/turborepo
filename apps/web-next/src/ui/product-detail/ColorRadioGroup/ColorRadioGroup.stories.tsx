import { useState } from 'react'

import type { Meta, StoryFn } from '@storybook/react'

// Mocks
import { MOCK_COLOR_RADIO_GROUP } from '@/mocks'

import ColorRadioGroup, { ColorRadioGroupProps } from '.'

const meta = {
  title: 'Components/ColorRadioGroup',
  tags: ['autodocs'],
  component: ColorRadioGroup,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<ColorRadioGroupProps>

export default meta

const Template: StoryFn<ColorRadioGroupProps> = (args) => {
  const [valueColor, setValueColor] = useState(args.valueColor)

  return (
    <ColorRadioGroup
      {...args}
      valueColor={valueColor}
      setValueColor={setValueColor}
    />
  )
}

export const Default = Template.bind({})
Default.args = {
  items: MOCK_COLOR_RADIO_GROUP,
  valueColor: MOCK_COLOR_RADIO_GROUP[0].color,
}
