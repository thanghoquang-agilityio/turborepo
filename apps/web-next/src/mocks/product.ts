import { ProductVariantModel } from '@/types'

import { MOCK_IMAGE } from './image'

// New flattened product variants for testing
export const MOCK_PRODUCT_VARIANTS: ProductVariantModel[] = [
  {
    id: 1,
    color: 'slate blue',
    size: 'small,medium,large,extra large,XXL',
    stock: 20,
    tax: 5,
  },
  {
    id: 2,
    color: 'blue',
    size: 'small,medium,large,extra large,XXL',
    stock: 20,
    tax: 5,
  },
]

export const MOCK_PRODUCT = {
  id: '1',
  image: MOCK_IMAGE,
  name: 'TDX Sinkers',
  variants: 5,
  price: 675.0,
  star: 5,
  comments: 121,
  description: '',
  discount: 0,
  likes: 11,
  brand: 'MOCK BRAND',
  productVariants: {
    data: MOCK_PRODUCT_VARIANTS,
  },
}

export const MOCK_PRODUCTS_RESPONSE = [
  {
    id: '1',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '2',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '3',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '4',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: MOCK_PRODUCT_VARIANTS,
      },
      image: MOCK_IMAGE,
    },
  },
]
