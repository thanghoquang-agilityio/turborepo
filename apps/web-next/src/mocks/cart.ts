import { CartItemResponse, LegacyCartItemResponse } from '@/types'
import { MOCK_PRODUCT_VARIANTS } from './product'

export const MOCK_CART_ITEM = {
  id: '1',
  variantId: '1',
  name: 'Embrace Sideboard',
  color: 'wheat',
  size: 'small',
  quantityDefault: 2,
  stock: 4,
  price: 53,
  star: 1,
  comments: 1,
  tax: 1,
  discount: 1,
}

// New flattened cart response structure
export const MOCK_CART_RESPONSE: CartItemResponse[] = [
  {
    id: 1,
    documentId: 'cart-item-1',
    size: 'small',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[0],
    userId: 1,
  },
  {
    id: 2,
    documentId: 'cart-item-2',
    size: 'large',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[1],
    userId: 1,
  },
]

// Legacy cart response for backward compatibility
export const MOCK_CART_RESPONSE_LEGACY: LegacyCartItemResponse[] = [
  {
    id: '1',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '2,10,0,23,20',
            tax: 12,
          },
        },
      },
      size: 'small',
      quantity: 2,
    },
  },
  {
    id: '2',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '20,10,20,23,20',
            tax: 12,
          },
        },
      },
      size: 'large',
      quantity: 2,
    },
  },
]

export const MOCK_ORDER_SUMMARY = {
  subTotal: 396,
  itemTotal: 3,
}
