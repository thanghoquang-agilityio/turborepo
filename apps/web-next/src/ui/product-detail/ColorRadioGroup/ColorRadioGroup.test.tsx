import { render } from '@testing-library/react'

import { MOCK_COLOR_RADIO_GROUP } from '@/mocks'

import ColorRadioGroup from '.'

const mockSetValueColor = jest.fn()

const renderComponent = () =>
  render(
    <ColorRadioGroup
      items={MOCK_COLOR_RADIO_GROUP}
      valueColor={MOCK_COLOR_RADIO_GROUP[0].color}
      setValueColor={mockSetValueColor}
    />
  )

describe('ColorRadioGroup', () => {
  beforeEach(() => {
    const mockIntersectionObserver = jest.fn()
    mockIntersectionObserver.mockReturnValue({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    })
    window.IntersectionObserver = mockIntersectionObserver
  })

  it('should render with default props', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
