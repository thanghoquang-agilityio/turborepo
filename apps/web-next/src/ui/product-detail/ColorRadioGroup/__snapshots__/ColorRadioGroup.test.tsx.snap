// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorRadioGroup should render with default props 1`] = `
<div>
  <div
    aria-label="Choose a Color"
    aria-orientation="horizontal"
    class="relative flex flex-col sm:gap-4 gap-2"
    id="react-aria-:r0:"
    role="radiogroup"
  >
    <span
      class="relative text-base text-foreground-300 font-medium"
    >
      Choose a Color
    </span>
    <div
      class="flex flex-col flex-wrap data-[orientation=horizontal]:flex-row sm:gap-4 gap-2"
      data-orientation="horizontal"
      role="presentation"
    >
      <label
        aria-label="Radio Color"
        class="group relative inline-flex cursor-pointer tap-highlight-transparent justify-center items-center max-w-[64px] sm:w-[64px] w-[54px] sm:h-[64px] h-[54px] m-0 capitalize rounded-full border-[3px] border-success-200 p-0"
        data-selected="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":r9: react-aria-:r3:"
            aria-label="Radio Color"
            aria-labelledby=":r8:"
            checked=""
            name="react-aria-:r5:"
            tabindex="0"
            type="radio"
            value="yellow green"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-medium box-border border-default outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:border-primary group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex bg-success-200 group-data-[hover-unselected=true]:bg-success-200 sm:w-[50px] sm:h-[50px] w-[40px] h-[40px] hover:opacity-80 p-0 m-0 cursor-pointer rounded-full border-none"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground w-2 h-2 transition-transform-opacity motion-reduce:transition-none bg-background-100 hidden"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset] absolute sm:top-[19px] top-[16px] sm:left-[10px] left-[6px] sm:w-6 sm:h-5 w-5 h-4"
        >
          <span
            class="relative text-foreground select-none text-medium transition-colors motion-reduce:transition-none"
            id=":r8:"
          >
            <div>
              <svg
                class="text-background-100"
                fill="none"
                viewBox="0 0 24 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M23.2536 0.252736C23.8044 0.668905 23.9135 1.4528 23.4973 2.00361L10.7473 18.8786C10.5376 19.1563 10.2215 19.3339 9.87529 19.3688C9.52904 19.4037 9.18393 19.2926 8.92299 19.0624L0.422988 11.5624C-0.094667 11.1056 -0.144037 10.3157 0.312718 9.79804C0.769472 9.28039 1.55939 9.23102 2.07704 9.68777L9.56573 16.2954L21.5027 0.496529C21.9188 -0.0542836 22.7027 -0.163433 23.2536 0.252736Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </span>
        </div>
      </label>
      <label
        aria-label="Radio Color"
        class="group relative inline-flex cursor-pointer tap-highlight-transparent p-2 justify-center items-center max-w-[64px] sm:w-[64px] w-[54px] sm:h-[64px] h-[54px] m-0 capitalize rounded-full border-[3px] border-background-100"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":rd: react-aria-:r3:"
            aria-label="Radio Color"
            aria-labelledby=":rc:"
            name="react-aria-:r5:"
            tabindex="-1"
            type="radio"
            value="violet"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-medium box-border border-default outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:border-primary group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex bg-danger-200 group-data-[hover-unselected=true]:bg-danger-200 sm:w-[50px] sm:h-[50px] w-[40px] h-[40px] hover:opacity-80 p-0 m-0 cursor-pointer rounded-full border-none"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground w-2 h-2 transition-transform-opacity motion-reduce:transition-none bg-background-100 hidden"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset] absolute sm:top-[19px] top-[16px] sm:left-[10px] left-[6px] sm:w-6 sm:h-5 w-5 h-4"
        />
      </label>
      <label
        aria-label="Radio Color"
        class="group relative inline-flex cursor-pointer tap-highlight-transparent p-2 justify-center items-center max-w-[64px] sm:w-[64px] w-[54px] sm:h-[64px] h-[54px] m-0 capitalize rounded-full border-[3px] border-background-100"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":rh: react-aria-:r3:"
            aria-label="Radio Color"
            aria-labelledby=":rg:"
            name="react-aria-:r5:"
            tabindex="-1"
            type="radio"
            value="wheat"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-medium box-border border-default outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:border-primary group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex bg-warning-300 group-data-[hover-unselected=true]:bg-warning-300 sm:w-[50px] sm:h-[50px] w-[40px] h-[40px] hover:opacity-80 p-0 m-0 cursor-pointer rounded-full border-none"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground w-2 h-2 transition-transform-opacity motion-reduce:transition-none bg-background-100 hidden"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset] absolute sm:top-[19px] top-[16px] sm:left-[10px] left-[6px] sm:w-6 sm:h-5 w-5 h-4"
        />
      </label>
      <label
        aria-label="Radio Color"
        class="group relative inline-flex cursor-pointer tap-highlight-transparent p-2 justify-center items-center max-w-[64px] sm:w-[64px] w-[54px] sm:h-[64px] h-[54px] m-0 capitalize rounded-full border-[3px] border-background-100"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":rl: react-aria-:r3:"
            aria-label="Radio Color"
            aria-labelledby=":rk:"
            name="react-aria-:r5:"
            tabindex="-1"
            type="radio"
            value="slate blue"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-medium box-border border-default outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:border-primary group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex bg-secondary-400 group-data-[hover-unselected=true]:bg-secondary-400 sm:w-[50px] sm:h-[50px] w-[40px] h-[40px] hover:opacity-80 p-0 m-0 cursor-pointer rounded-full border-none"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground w-2 h-2 transition-transform-opacity motion-reduce:transition-none bg-background-100 hidden"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset] absolute sm:top-[19px] top-[16px] sm:left-[10px] left-[6px] sm:w-6 sm:h-5 w-5 h-4"
        />
      </label>
      <label
        aria-label="Radio Color"
        class="group relative inline-flex cursor-pointer tap-highlight-transparent p-2 justify-center items-center max-w-[64px] sm:w-[64px] w-[54px] sm:h-[64px] h-[54px] m-0 capitalize rounded-full border-[3px] border-background-100"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":rp: react-aria-:r3:"
            aria-label="Radio Color"
            aria-labelledby=":ro:"
            name="react-aria-:r5:"
            tabindex="-1"
            type="radio"
            value="salmon - dark green"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-medium box-border border-default group-data-[hover-unselected=true]:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:border-primary group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex bg-gradient-conic sm:w-[50px] sm:h-[50px] w-[40px] h-[40px] hover:opacity-80 p-0 m-0 cursor-pointer rounded-full border-none"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground w-2 h-2 transition-transform-opacity motion-reduce:transition-none bg-background-100 hidden"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset] absolute sm:top-[19px] top-[16px] sm:left-[10px] left-[6px] sm:w-6 sm:h-5 w-5 h-4"
        />
      </label>
    </div>
  </div>
</div>
`;
